src/sgxenv.mk

```diff
@@ -128,14 +128,28 @@ SGX_CXXFLAGS_T := $(SGX_CFLAGS_T) -std=c++11 -nostdinc++ -I$(SGX_SDK)/include/li
 # --nostdlib: Do not use system startup files or libraries when linking. Thus, 
 #  only the libaries that are explictly specified on the command line are 
 #  linked.
+ifneq ($(SGX_MODE), HW)
+SGX_LFLAGS_T = $(SGX_COMMON_CFLAGS) -nostdlib -L$(SGX_LIBRARY_PATH) $(_Other_Link_Flags) \
+	-Wl,--whole-archive -l$(Trts_Library_Name) -Wl,--no-whole-archive \
+	-Wl,--start-group -lsgx_tcxx -lsgx_tstdc -l$(Crypto_Library_Name) -l$(Service_Library_Name) $(_Other_Enclave_Libs) -Wl,--end-group \
+	-Wl,-Bstatic -Wl,-Bsymbolic -Wl,--no-undefined \
+	-Wl,-pie,-eenclave_entry -Wl,--export-dynamic  \
+	-Wl,--defsym,__ImageBase=0 \
+	-Wl,--gc-sections \
+	-Wl,--version-script=Enclave.lds
+else
+# enable tcmalloc in HW mode only
 SGX_LFLAGS_T = $(SGX_COMMON_CFLAGS) -nostdlib -L$(SGX_LIBRARY_PATH) $(_Other_Link_Flags) \
 	-Wl,--whole-archive -l$(Trts_Library_Name) -Wl,--no-whole-archive \
+	-Wl,--whole-archive -lsgx_tcmalloc -Wl,--no-whole-archive \
```

Comment:
Having two definitions of `SGX_LFLAGS_T` that span multiple lines with only one line difference is not ideal: too much code duplication and very hard to identify the difference. For better readability and maintainability, the modification needs to be simplified (and it can be).